internal/services/storage/storage_share_resource.go

```diff
@@ -143,239 +150,424 @@ func resourceStorageShare() *pluginsdk.Resource {
+			}
 
-	protocol := shares.ShareProtocol(d.Get("enabled_protocol").(string))
-	if protocol == shares.NFS {
-		// Only FileStorage (whose sku tier is Premium only) storage account is able to have NFS file shares.
-		// See: https://learn.microsoft.com/en-us/azure/storage/files/storage-files-quick-create-use-linux#applies-to
-		if account.Kind != storageaccounts.KindFileStorage {
-			return fmt.Errorf("NFS File Share is only supported for Storage Account with kind %q but got `%s`", string(storageaccounts.KindFileStorage), account.Kind)
+			return resourceStorageShareRead(d, meta)
 		}
 	}
 
-	// The files API does not support bearer tokens (@manicminer, 2024-02-15)
-	client, err := storageClient.FileSharesDataPlaneClient(ctx, *account, storageClient.DataPlaneOperationSupportingOnlySharedKeyAuth())
+	accountId, err := commonids.ParseStorageAccountID(d.Get("storage_account_id").(string))
 	if err != nil {
-		return fmt.Errorf("building File Share Client: %v", err)
+		return err
 	}
 
-	exists, err := client.Exists(ctx, shareName)
+	id := fileshares.NewShareID(accountId.SubscriptionId, accountId.ResourceGroupName, accountId.StorageAccountName, d.Get("name").(string))
+
+	existing, err := sharesClient.Get(ctx, id, fileshares.DefaultGetOperationOptions())
 	if err != nil {
-		return fmt.Errorf("checking for existing %s: %v", id, err)
+		if !response.WasNotFound(existing.HttpResponse) {
+			return fmt.Errorf("checking for existing %q: %v", id, err)
+		}
 	}
-	if exists != nil && *exists {
+	if !response.WasNotFound(existing.HttpResponse) {
 		return tf.ImportAsExistsError("azurerm_storage_share", id.ID())
 	}
 
-	log.Printf("[INFO] Creating Share %q in Storage Account %q", shareName, accountName)
-	input := shares.CreateInput{
-		QuotaInGB:       quota,
-		MetaData:        metaData,
-		EnabledProtocol: protocol,
+	payload := fileshares.FileShare{
+		Properties: &fileshares.FileShareProperties{
+			EnabledProtocols:  pointer.To(fileshares.EnabledProtocols(d.Get("enabled_protocol").(string))),
+			Metadata:          pointer.To(ExpandMetaData(d.Get("metadata").(map[string]interface{}))),
+			ShareQuota:        pointer.To(int64(d.Get("quota").(int))),
+			SignedIdentifiers: expandStorageShareACLs(d.Get("acl").(*pluginsdk.Set).List()),
+		},
 	}
 
-	if accessTier := d.Get("access_tier").(string); accessTier != "" {
-		tier := shares.AccessTier(accessTier)
-		input.AccessTier = &tier
+	if sharedAccessTier, ok := d.GetOk("access_tier"); ok && sharedAccessTier.(string) != "" {
+		payload.Properties.AccessTier = pointer.To(fileshares.ShareAccessTier(sharedAccessTier.(string)))
 	}
 
-	if err = client.Create(ctx, shareName, input); err != nil {
+	pollerType := custompollers.NewStorageShareCreatePoller(sharesClient, id, payload)
+	poller := pollers.NewPoller(pollerType, 5*time.Second, pollers.DefaultNumberOfDroppedConnectionsToAllow)
+
+	if err = poller.PollUntilDone(ctx); err != nil {
 		return fmt.Errorf("creating %s: %v", id, err)
 	}
 
 	d.SetId(id.ID())
 
```

Comment:
Not sure if I've overlooked it, but we're not setting the `acl` property in the new RM behaviour for storage shares and it hasn't been deprecated